docs/Server.md

```diff
@@ -228,6 +228,8 @@ By default, value equal to `true`, routes are registered as case sensitive. That
 
 By setting `caseSensitive` to `false`, all paths will be matched as lowercase, but the route parameters or wildcards will maintain their original letter casing.
 
+This option also applies to query strings. When `caseSensitive` is set to `false`, `/foo?query=bar` is equivalent to `/foo?QUERY=bar` and the other variants (`qUerY`, `Query`, etc.). Properties with case variations are converted into a single argument. e.g. (`foo=bar1&Foo=bar2&foO=bar3` gets converted to `{foo: ['bar1', 'bar2', 'bar3']}`
```

Comment:
As @jsumners mentioned [here](https://github.com/fastify/fastify/pull/2649#discussion_r511913091) this is breaking change